turbinia/state_manager.py

```diff
@@ -327,6 +334,256 @@ def write_new_task(self, task):
+
+    Args:
+      evidence_id (str): The ID of the stored evidence.
+
+    Returns:
+      evidence_dict (dict): Dict containing evidence attributes. 
+    """
+    evidence_key = ':'.join(('TurbiniaEvidence', evidence_id))
+    evidence_dict = {}
+    for attribute_name in self.iterate_attribute_names(evidence_key):
+      evidence_dict[attribute_name] = self.get_attribute(
+          evidence_key, attribute_name)
+    return evidence_dict
+
+  def get_evidence_summary(
+      self, sort: str = None, output: str = 'keys') -> dict | int:
+    """Gets a summary of all evidences.
+
+    Args:
+      sort (str): Name of the evidence attribute used to sort the output.
+      output (str): Value to be output by the function (keys | values | count).
+
+    Returns:
+      summary (dict | int): Object containing evidences. 
+    """
+    if output == 'count' and not sort:
+      return sum(1 for _ in self.iterate_keys('Evidence'))
+    summary = {} if sort else []
+    for evidence_key in self.iterate_keys('Evidence'):
+      evidence_dictionary = self.get_evidence_data(evidence_key.split(':')[1])
+      stored_value = evidence_dictionary if output == 'values' else (
+          evidence_key)
+      if sort:
+        attribute_value = evidence_dictionary.get(sort, None)
+        if attribute_value not in summary:
+          summary[attribute_value] = [stored_value] if output != 'count' else 1
+        elif output == 'count':
+          summary[attribute_value] += 1
+        else:
+          summary[attribute_value].append(stored_value)
+        continue
+      summary.append(stored_value)
+    return summary
+
+  def query_evidence(
+      self, attribute_name: str, value: Any,
+      output: str = 'keys') -> dict | list | int:
+    """Queries for evidences with the specified attribute matching the passed
+    value.
+
+    Args:
+      attribute_name (str): Name of the attribute containing the value.
+      value (Any): Value stored in the attribute.
+      output (str): Value to be output by the function (keys | values | count).
+
+    Returns:
+      query_result (dict | list | int): Result of the query. 
+    """
+    keys = []
+    for evidence_key in self.iterate_keys('Evidence'):
+      if attribute_value := self.get_attribute(evidence_key, attribute_name):
+        if (attribute_name == 'tasks' and value in attribute_value
+           ) or attribute_value == value or str(attribute_value) == str(value):
+          keys.append(evidence_key)
+    if output == 'values':
+      return [self.get_evidence_data(key.split(':')[1]) for key in keys]
+    elif output == 'count':
+      return len(keys)
+    return keys
+
+  def get_evidence_key_by_hash(self, file_hash: str):
+    """Gets the evidence key given its hash.
+
+    Args:
+      file_hash (str): The hash of the stored evidence.
+
+    Returns:
+      key (str): Key of the stored evidence. 
+    """
+    if file_hash:
+      return self.get_attribute('TurbiniaEvidenceHashes', file_hash, False)
```

Comment:
This might raise a TurbiniaException, should it be handled here or in the method that calls get_evidence_key_by_hash?